examples/typescript/__tests__/calc.test.ts

```diff
@@ -7,8 +7,8 @@ jest.mock('../memory');
 jest.mock('../sub');
 jest.mock('../sum');
 
-const mockSub = sub as jest.MockedFunction<typeof sub>;
-const mockSum = sum as jest.MockedFunction<typeof sum>;
+const mockSub = jest.mocked(sub);
+const mockSum = jest.mocked(sum);
 const MockMemory = Memory as jest.MockedClass<typeof Memory>;
```

Comment:
@mrazauskas Hiya! I cannot use `jest.mocked` here as it returns `jets.MockedObject<typeof Memory>` instead of `MockedClass`. This happens both with `@jest/globals` and `@types/jest` (whom I think just copied the implementation from `@jest/globals`). Any ideas on how we can make `jest.mocked` work for classes as well?

Reply:
Yep. I encountered this while working on #12727. That PR is fixing this issue. In general the work on types is finished in that PR, it just needs more tests. In this case these are rather complex ones. Indeed it would be nice to land it for Jest 29. I will try to push forward with the tests, if you can wait with the release for a couple of days (;

Reply:
Awesome, thanks! Happy to wait with the release until next week, no rush 🙂